frontend/app/work_packages/services/work-package-field-service.js

```diff
@@ -323,6 +335,9 @@ module.exports = function(
   function format(workPackage, field) {
     var schema = getSchema(workPackage);
     if (field === 'date') {
+      if(workPackage.embedded.type.props.name === 'Milestone') {
+        return workPackage.props.startDate;
```

Comment:
@Azure7111 we need to know that the type in the form response is a Milestone because we first do a post request to the form endpoint, and it returns only a link to the type. I am not sure if making another blocking request to the type endoint on each type change is worth it